src/pages/academy/grading/subcomponents/GradingBadges.tsx

```diff
@@ -1,28 +1,102 @@
 import { Icon } from '@blueprintjs/core';
 import { IconNames } from '@blueprintjs/icons';
-import { ColumnFilter } from '@tanstack/react-table';
-import { Badge } from '@tremor/react';
+import classNames from 'classnames';
+import React from 'react';
 import { ProgressStatus, ProgressStatuses } from 'src/commons/assessment/AssessmentTypes';
+import { ColumnFilter } from 'src/features/grading/GradingTypes';
+import classes from 'src/styles/Grading.module.scss';
+import badgeClasses from 'src/styles/GradingBadges.module.scss';
+
+declare const sizeValues: readonly ['xs', 'sm', 'md', 'lg', 'xl'];
+declare type Size = (typeof sizeValues)[number];
+
+type BadgeProps = {
+  text: string;
+  /** First color is bg, second is text. Refer to {typeof AVAILABLE_COLORS} */
+  color?: readonly [string, string];
+  size?: Size;
+  icon?: () => React.ReactNode;
```

Comment:
Let's just pass the node directly. That way no need for optional chaining as well since `undefined` is a valid React node.

Reply:
Resolved in 9bc265690e6cda0a68b470d823b21273a5b90b6d